### PR TITLE
Landing page for EFH, M&E and others

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/3bap2r8

### Notes

- Adds `Config` entity with fallback information. It's persisted in the dataStore. Example:
  - http://localhost:8080/dhis-web-datastore/index.html
  - namespace: `landing-page-app`
  - key: `config`
  - JSON value: `{"fallbackUrl": "/dhis-web-dashboard/index.action", "defaultProgramme": "international-projects"}`.

First, it tries to match a programme with `defaultProgramme`. If not set or no match, it uses `fallbackUrl` (defaults to dashboard app).
  
### :art: Screenshots
Screenshot 2022-08-19 at 13 43 33

### CODE REVIEW

File: src/data/logic/redirection.ts [hunk 1]
```diff
@@ -136,6 +145,16 @@ export const buildAvailableConfigurations = (version: number): Configuration[] =
...
+        programme: "international-projects",
+        title: i18n.t("International Projects"),
+        description: i18n.t("Landing Page for International Projects"),
+        userGroupIds: [EFH_ADMIN, EFH_DASHBOARD, EFH_DATA_ENTRY, EFH_MIO, DATA_MONITORING],
+        page: InternationalLandingPage,
+        header: internationalHeader,
+        data: internationalData,
+        icon: "img/icon.png",
+    },
 ];
```
Comment (tokland) [inline src/data/logic/redirection.ts:158]: This entry makes a redirect to the IP Landing page when some group is matched. But from the task description, we need something more: render the project monitoring app for DATA_MONITORING only-match and EFH app for the other groups only-match. Only if we belong to both groups we render the IP landing page. @adrian Note that "no group" scenario is not generically supported by the app; in that case, it renders the root landing page.

Comment (deeonwuli) [reply src/data/logic/redirection.ts:158]: @tokland i'm not entirely sure how to go about this. is this similar to what is done in the src/domain/models/hepatitis/Hepatitis.js file?

Comment (tokland) [reply src/data/logic/redirection.ts:158]: We've discussed the approach with @adrianq, it's not trivial, he'll contact you on how to tackle this.

Comment (deeonwuli) [reply src/data/logic/redirection.ts:158]: okay. thanks @tokland

**Comment by tokland** [comment]
[ ] Add config entity with fallback info persisted in the dataStore

**Review by tokland — APPROVED** [review]

**Review by adrianq — APPROVED** [review]
@deeonwuli @tokland Thanks for the PR! It is working like a charm

@idelcano @ifoche As discussed I am adding the two of you as reviewers as this PR may have an impact on the WHO current use case. We are introducing the following relevant changes:
- Before, every superuser was sent to the main landing page. Now, only the user belonging to the user group "IT_MAINTENANCE_TEAM"
- We have added a new feature that should have no impact on WHO. Before, if the user was not part of any user group and he was not a superuser (no condition was matched) he was sent to the dashboard app (dhis-web-dashboard/index.action). Now, we first look for two potential keys in the datastore:
```
namespace: landing-page-app
key: config
JSON value: {"fallbackUrl": "/dhis-web-dashboard/index.action", "defaultProgramme": "international-projects"}
```
These settings allow you to define a fallback URL or landing page module. If these settings cannot be found in the datastore it will send the user to the dashboard app (as before). There is no way from the GUI to add the settings to the datastore. You have to do it manually.

Hope it is clear but let me know otherwise

**Review by ifoche — APPROVED** [review]
Thanks @deeonwuli @tokland @adrianq !! working like a charm

**Pull request merged**